src/Language/PureScript/TypeChecker.hs

```diff
@@ -205,9 +206,10 @@ typeCheckAll moduleName _ = traverse go
       addDataType moduleName dtype name args' dctors ctorKind
     return $ DataDeclaration dtype name args dctors
   go (d@(DataBindingGroupDeclaration tys)) = do
-    warnAndRethrow (addHint ErrorInDataBindingGroup) $ do
-      let syns = mapMaybe toTypeSynonym tys
-      let dataDecls = mapMaybe toDataDecl tys
+    let syns = mapMaybe toTypeSynonym tys
+    let dataDecls = mapMaybe toDataDecl tys
+    let bindingGroupNames = (syns^..traverse._1) ++ (dataDecls^..traverse._2)
```

Comment:
I don't think I can use `map snd` as `dataDecls` elements are triples.
I can switch to `map (\(_, n, _) -> n) dataDecls` if you prefer?  There is similar code on the next few lines.
Using `nub` sounds good.